tests/api/ocsp.c

```diff
@@ -0,0 +1,539 @@
+}
+
+int test_ocsp_basic_verify(void)
+{
+    EXPECT_DECLS;
+    WOLF_STACK_OF(WOLFSSL_X509) * certs;
+    OcspResponse* response = NULL;
+    WOLFSSL_X509_STORE* store;
+    const unsigned char* ptr;
+    DecodedCert cert;
+    int ret;
+
+    wc_InitDecodedCert(&cert, ocsp_responder_cert_pem,
+        sizeof(ocsp_responder_cert_pem), NULL);
+    ret = wc_ParseCert(&cert, CERT_TYPE, 0, NULL);
+    ExpectIntEQ(ret, 0);
+
+    /* just decoding */
+    ptr = (const unsigned char*)resp;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp));
+    ExpectPtrNE(response, NULL);
+    ExpectIntEQ(response->responseStatus, 0);
+    ExpectIntEQ(response->responderIdType, OCSP_RESPONDER_ID_NAME);
+    ExpectBufEQ(response->responderId.nameHash, cert.subjectHash,
+        OCSP_DIGEST_SIZE);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* responder Id by key hash */
+    ptr = (const unsigned char*)resp_rid_bykey;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_rid_bykey));
+    ExpectPtrNE(response, NULL);
+    ExpectIntEQ(response->responseStatus, 0);
+    ExpectIntEQ(response->responderIdType, OCSP_RESPONDER_ID_KEY);
+    ExpectBufEQ(response->responderId.keyHash, cert.subjectKeyHash,
+        OCSP_DIGEST_SIZE);
+    wc_FreeDecodedCert(&cert);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* decoding with no embedded certificates */
+    ptr = (const unsigned char*)resp_nocert;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_nocert));
+    ExpectPtrNE(response, NULL);
+    ExpectIntEQ(response->responseStatus, 0);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* decoding an invalid response */
+    ptr = (const unsigned char*)resp_bad;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_bad));
+    ExpectPtrEq(response, NULL);
+
+    ptr = (const unsigned char*)resp;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp));
+    ExpectPtrNE(response, NULL);
+    /* no verify signer certificate */
+    ret = wolfSSL_OCSP_basic_verify(response, NULL, NULL, OCSP_NOVERIFY);
+    ExpectIntEQ(ret, WOLFSSL_SUCCESS);
+    /* verify that the signature is checked */
+    response->sig[0] ^= 0xff;
+    ret = wolfSSL_OCSP_basic_verify(response, NULL, NULL, OCSP_NOVERIFY);
+    ExpectIntEQ(ret, WOLFSSL_FAILURE);
+    wolfSSL_OCSP_RESPONSE_free(response);
+
+    /* populate a store with root-ca-cert */
+    ret = test_ocsp_create_x509store(&store, root_ca_cert_pem,
+        sizeof(root_ca_cert_pem));
+    ExpectIntEQ(ret, TEST_SUCCESS);
+
+    /* populate a WOLF_STACK_OF(WOLFSSL_X509) with responder certificate */
+    ret = test_create_stack_of_x509(&certs, ocsp_responder_cert_pem,
+        sizeof(ocsp_responder_cert_pem));
+    ExpectIntEQ(ret, TEST_SUCCESS);
+
+    /* cert not embedded, cert in certs, validated using store */
+    ptr = (const unsigned char*)resp_nocert;
+    response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_nocert));
+    ExpectPtrNE(response, NULL);
```

Comment:
`ExpectNotNull`. Correct others as well.

Reply:
It can even be on the same line
```suggestion
    ExpectNotNull(response = wolfSSL_d2i_OCSP_RESPONSE(NULL, &ptr, sizeof(resp_nocert)));
```